packages/dev/core/src/Meshes/geometry.ts

```diff
@@ -491,7 +491,7 @@ export class Geometry implements IGetSetVerticesData {
      * @returns true if data is present
      */
     public isVerticesDataPresent(kind: string): boolean {
-        if (!this._vertexBuffers) {
+        if (!this._vertexBuffers || !this.getVerticesData(kind)) {

```

Comment:
I'm not for this change because the call to `getFloatData` that is done in `getVerticesData` may create a new buffer and copy the data into it whereas we don't need/use it.

I think it's the code below which is wrong:
```javascript
const box = BABYLON.MeshBuilder.CreateBox("box", { size: 1 }, scene);
box.getVertexBuffer(BABYLON.VertexBuffer.PositionKind)?.dispose();
box.bakeTransformIntoVertices(BABYLON.Matrix.Scaling(-1, -1, -1));
```
You should not dispose a buffer that is in use by a mesh without the mesh knowing. You should dispose this buffer with the right method, which is `removeVerticesData`:
```javascript
const box = BABYLON.MeshBuilder.CreateBox("box", { size: 1 }, scene);
box.removeVerticesData(BABYLON.VertexBuffer.PositionKind);
box.bakeTransformIntoVertices(BABYLON.Matrix.Scaling(-1, -1, -1));
```

Reply:
I agree. The test is still nice to have still (but with the correct test code ;))

Reply:
Thaks for your feedback!
I fix the `isVerticesDataPresent` method and test case, check it, please.